### PR TITLE
doc: add page about creating modules

### PR DESCRIPTION
**Pull request by authmillenon**
Moar doc :-)

### CODE REVIEW

File: doc/doxygen/src/creating-modules.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+Modules can be defined outside `RIOTBASE`. In addition to add it to `USEMODULE`
+the user needs to add the path to the module to `EXTERNAL_MODULES` and add the
+include path to the API definitions to `INCLUDES`.
+
+Pseudomodules                                                  {#pseudomodules}
+=============
+Pseudomodules are modules that do not have any code. Their main use cases are
+to provide base information for dependencies to other modules or information to
+the code base via the `MODULE_<MODULENAME>` macro. Pseudomodules can provide
+header files too, if need be. To create a pseudomodule just add its name to the
```
Comment (PeterKietzmann) [inline doc/doxygen/src/creating-modules.md]: > if need to be

Stumbled upon that. How about "if there is a need for it" or similar

File: doc/doxygen/src/creating-modules.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+================
+The general structure                                  {#the-general-structure}
+=====================
+Like @ref creating-an-application "applications", modules are directories
+containing source files and a Makefile. Additionally their API can be defined
+in one or more header files, residing in the include path of their
+super-module.
+
+E.g. the @ref sys_shell module is implemented in `sys/shell` and defines its
+API in `sys/include/shell.h`.
```
Comment (PeterKietzmann) [inline doc/doxygen/src/creating-modules.md]: Maybe it's nice to mention  which kinds of modules exist. Not like a complete list, but to inform that e.g. drivers are also modules

**Comment by PeterKietzmann** [comment]
I think it could be interesting to inform that the Makefile.dep handles some modules dependencies.

**Comment by PeterKietzmann** [comment]
Again: Thanks Martine :-) !

File: doc/doxygen/src/creating-modules.md (hunk 2 above)
Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: Which one would that be? Also this page is primarily about *creating* modules, not using them ;-).

File: doc/doxygen/src/creating-modules.md (hunk 1 above)
Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: "if need be" is a correct phrasing: http://www.oxforddictionaries.com/de/definition/englisch/if-need-be

Comment (Lotterleben) [inline doc/doxygen/src/creating-modules.md]: +1, also “if need be” seems more concise to me.

**Comment by Lotterleben** [comment]
Would it make sense to add a short sentence about what Modules actually are, maybe something along the lines of “Modules add a specific set of features to the RIOT base kernel” (or something o.o) or is that out of place here?

File: doc/doxygen/src/creating-modules.md (hunk 2 above)
Comment (Lotterleben) [inline doc/doxygen/src/creating-modules.md]: Having a (partial) list might help people to do their own digging for (good) examples...

Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: I'll add one if I get one. Up until now I got only: add some list ;-)

**Comment by authmillenon** [comment]
They do not add to the kernel, but I can try to formulate a description:

> Modules in RIOT are well-defined units of code that provide a set of features to your application.

**Comment by Lotterleben** [comment]
True. I like that sentence. :)

File: doc/doxygen/src/creating-modules.md (hunk 2 above)
Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: After thinking about it, I think a better place for such list is the "Creating an application" page. I'm assuming here that most people that create a module at least worked a little bit with applications so they would have already seen that list. What do you guys think?

**Comment by authmillenon** [comment]
Rebased and addressed comments (added section about definition of modules including clarifications that drivers and to an extend CPU and board ports [**disclaimer:** can someone like @haukepetersen fact-prove that ;-)] are modules, too; added section about `Makefile.dep`)

File: doc/doxygen/src/creating-modules.md (hunk 2 above)
Comment (Lotterleben) [inline doc/doxygen/src/creating-modules.md]: I'm starting to think that The List(tm) might be beneficial in both places :D

File: doc/doxygen/src/creating-modules.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+Creating modules                                            {#creating-modules}
+================
+Modules in RIOT are well-defined units of code that provide a set of features
+to your application. This includes also drivers and to a further extend ports
```
Comment (Lotterleben) [inline doc/doxygen/src/creating-modules.md]: I think it should be "to a certain extent"

File: doc/doxygen/src/creating-modules.md (hunk 2 above)
Comment (PeterKietzmann) [inline doc/doxygen/src/creating-modules.md]: The only information that I was/am missing here was a hint that a module must not only be a system or kernel module but also a e.g. driver. IMO no need for a complete list here. The diver porting guide will also state that a driver might become a module

Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: Got it. Is it now better? @Lotterleben are you alright with this compromise?

File: doc/doxygen/src/creating-modules.md (hunk 3 above)
Comment (authmillenon) [inline doc/doxygen/src/creating-modules.md]: Fixed

**Comment by authmillenon** [comment]
Fixed more comments. Now let's merge this so I can open a last minute release PR ;-)

**Comment by Lotterleben** [comment]
ACK :) Could you squash please?

**Comment by authmillenon** [comment]
Squashed.

**Pull request merged**